olp-cpp-sdk-dataservice-read/tests/ConfigApiTest.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "generated/api/ConfigApi.h"
+
+#include <gmock/gmock.h>
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/client/HttpResponse.h>
+
+namespace {
+
+using namespace ::olp::dataservice::read;
+using namespace ::olp::client;
+using namespace ::testing;
+
+class ConfigApiTestable : public ConfigApi {
+ public:
+  using DoCallApiCallback = std::function<HttpResponse(
+      std::string, std::string, std::multimap<std::string, std::string>,
+      std::multimap<std::string, std::string>,
+      std::multimap<std::string, std::string>,
+      std::shared_ptr<std::vector<unsigned char>>, std::string,
+      CancellationContext)>;
+
+  explicit ConfigApiTestable(DoCallApiCallback callback)
+      : call_api_callback_(std::move(callback)) {}
+
+ protected:
+  HttpResponse DoCallApi(const OlpClient& client, std::string path,
+                         std::string method,
+                         std::multimap<std::string, std::string> query_params,
+                         std::multimap<std::string, std::string> header_params,
+                         std::multimap<std::string, std::string> form_params,
+                         std::shared_ptr<std::vector<unsigned char>> post_body,
+                         std::string content_type,
+                         CancellationContext context) override {
+    return call_api_callback_(std::move(path), std::move(method),
+                              std::move(query_params), std::move(header_params),
+                              std::move(form_params), std::move(post_body),
+                              std::move(content_type), context);
+  }
```

Comment:
It could be mocked method, with expectations in tests cases, did you tried it?

Reply:
Yes, I tried using the mocks first, but I stumbled on some problem with this usage... But I guess it was something with private methods. Anyway, yeah, I could have used GMock here.

But currently I removed those tests, so, this code is not relevant anymore